### DISCUSSION TITLE
Show module list in docs as a tree

### DISCUSSION THREAD

**Discussion by NicklasXYZ** [gleam-lang/gleam#1658] [Ideas & suggestions]
Suggestion: Make it possible to show the module list as a (hierarchical) tree.
If the module names are long it makes the list look slightly messy :)
E.g. see: https://hexdocs.pm/gleam_stats

**Comment by inoas** [comment]
https://github.com/gleam-lang/gleam/blob/main/compiler-core/templates/documentation_layout.html#L203-L208

How can I create the documentation locally so that I can impove this?

**Comment by NicklasXYZ** [comment]
If you're talking about the example I posted you could just clone the repo and build the docs, then serve them with a http server e.g. python -m http.server?

**Comment by NicklasXYZ** [comment]
Or are you referring to the doc template you're linking to?

**Comment by lpil** [comment]
What might this look like? I find most tree nav menus are very unwieldy and require a lot of clicking, and they make searching more difficult. We would need to ensure any new nav design can be used as quickly and easily as the current one.

**Reply by NicklasXYZ** [reply]
I was probably imagining something like the structure you get with e.g. `tree` or something like this:

![image](https://user-images.githubusercontent.com/18580183/172871832-204cac5d-5f5d-402f-bcb0-32bfad56443c.png)
